### DISCUSSION TITLE
Why the vocab size is 51865?

### DISCUSSION THREAD

**Discussion by HKAB** [openai/whisper#361] [Q&A] [answered]
I noticed the n_vocab size is: 51865. However, the tokenizer only have 50364 unique token.

```python
len(whisper.tokenizer.get_tokenizer('vi').tokenizer)
# 50364
```

```python
checkpoint['dims']
# {'n_mels': 80,
#  'n_vocab': 51865,
#  'n_audio_ctx': 1500,
#  'n_audio_state': 384,
#  'n_audio_head': 6,
#  'n_audio_layer': 4,
#  'n_text_ctx': 448,
#  'n_text_state': 384,
#  'n_text_head': 6,
#  'n_text_layer': 4}
```
Can you explain the output size of the model for me? It looks like that sometimes the model generate empty token which is out of tokenizer's vocabulary size.

**Comment by jianfch** [comment, marked as answer]
50364 and after are all timestamp tokens. 50364 is time 0s.

**Comment by daxaxelrod** [comment]
Is there anywhere we can view the entire lexicon map? Which int maps to which english token?

**Reply by HKAB** [reply]
simply using for loop from 1 to 50363 and use Whisper tokenizer to decode it.

**Reply by daxaxelrod** [reply]
Figured as such, wondering if i could save a repl session haha

**Comment by shuaijiang** [comment]
reference to this line  https://github.com/openai/whisper/blob/b38a1f20f4b23f3f3099af2c3e0ca95627276ddf/whisper/tokenizer.py#L344

`` *[f"<|{i * 0.02:.2f}|>" for i in range(1501)],

from 50364  to 51864， each is a timestamp from <|0.00|> to <|30.00|> with 0.02 interval
